src/config/src/meta/destinations.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2024 OpenObserve Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::fmt;
+
+use hashbrown::HashMap;
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub struct Destination {
+    pub id: String,
+    pub org_id: String,
+    pub name: String,
+    pub module: Module,
+}
+
+#[derive(Serialize, Debug, Deserialize, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum Module {
+    Alert {
+        template: String,
+        destination_type: DestinationType,
+    },
+    Pipeline {
+        pipeline_id: String,
+        endpoint: Endpoint,
+    },
+}
+
+#[derive(Serialize, Debug, Deserialize, Clone)]
+#[serde(tag = "type")]
+#[serde(rename_all = "snake_case")]
+pub enum DestinationType {
+    Http(Endpoint),
+    Email(Email),
+    Sns(AwsSns),
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Email {
+    pub recipients: Vec<String>,
+}
+
+#[derive(Serialize, Debug, PartialEq, Eq, Deserialize, Clone)]
+pub struct Endpoint {
+    pub url: String,
+    #[serde(default)]
+    pub method: HTTPType,
+    #[serde(default)]
+    pub skip_tls_verify: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub headers: Option<HashMap<String, String>>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct AwsSns {
+    pub sns_topic_arn: String,
+    pub aws_region: String,
+}
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize, ToSchema)]
+pub enum HTTPType {
+    #[default]
+    #[serde(rename = "post")]
+    POST,
+    #[serde(rename = "put")]
+    PUT,
+    #[serde(rename = "get")]
+    GET,
+}
```

Comment:
Nit:

I'm not 100% sure but I think clippy might complain about the acronym casing here. I think the convention that it suggests might be `HttpType`, `Post`, `Put`, `Get`.